packages/astro/src/default/utils/content.ts

```diff
@@ -347,6 +366,15 @@ async function getFilesRefList(pathToFolder: string): Promise<FilesRefList> {
   return [filesRef, filePaths];
 }
 
+function formatTemplateFile(filename: string) {
+  // compare files without leading "/" so that patterns like ["src/index.js"] match "/src/index.js"
```

Comment:
I'm not super familiar with micromatch, but this comment and the logic after make me think that maybe this is not the library that we want?

I think we should have `visibleFiles` entries follow a similar format to `git`'s `gitignore` entries. That would lower the learning curve of TutorialKit a lot and not necessitate as much trial and error for users.

In particular if you say:

```yaml

template:
  visibleFiles:
    - /foobar
```

It should not match `/test/foobar` but match a `/foobar` file in the template.

If you mean to match every file named `foobar` then you can only say `foobar` or `**/foobar`.

Reply:
I picked `micromatch` since it's already a transitive dependency from `fast-glob`.

`.gitignore` doesn't require prefixing all files with `/`? Meaning that:

```sh
$ touch new-file.ts
$ git status

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        new-file.ts

$ echo "new-file.ts" >> .gitignore 
$ git status

Changes not staged for commit:
        modified:   .gitignore
```

What `formatTemplateFile` here does, is that it allows you to defined `visibleFile: ["src/main.js"]` instead of `["/src/main.js"]`. Do we want to revert this `/` handling?

Reply:
Though I guess metadata's `focus` option does require `/` prefixes for all files.

Reply:
@Nemikolh here's small playground to check how it works:
https://stackblitz.com/~/edit/stackblitz-starters-efozpk?file=index.mjs
`node index.mjs` to run.

Reply:
I modified the example a bit to showcase what I mean: https://stackblitz.com/~/edit/stackblitz-starters-bq7pxo.

So basically when we have:

```js
[
  '/foobar/a.js',
  '/a.js',
]
```

If you have a `.gitignore` that says:

```
a.js
```

then it will ignore both files. But if it says `/a.js`, it will only ignore the second one.

Reply:
Oh looks like this is working! This might be good enough?

Reply:
cc @SamVerschueren

Reply:
Now using `basename: true`.